internal/archive/archive_test.go

```diff
@@ -371,22 +374,46 @@ func (s *httpSuite) TestArchiveLabels(c *C) {
 	}
 
 	_, err = archive.Open(&options)
-	c.Assert(err, IsNil)
+	c.Assert(err, ErrorMatches, `corrupted archive InRelease file: no Ubuntu section`)
+}
 
-	s.prepareArchiveAdjustRelease("jammy", "22.04", "amd64", []string{"main", "universe"}, setLabel("ThirdParty"))
+func (s *httpSuite) TestProArchives(c *C) {
+	setLabel := func(label string) func(*testarchive.Release) {
+		return func(r *testarchive.Release) {
+			r.Label = label
+		}
+	}
 
-	options = archive.Options{
-		Label:      "ubuntu",
-		Version:    "22.04",
-		Arch:       "amd64",
-		Suites:     []string{"jammy"},
-		Components: []string{"main", "universe"},
-		CacheDir:   c.MkDir(),
-		PubKeys:    []*packet.PublicKey{s.pubKey},
+	credsDir := c.MkDir()
+	restore := fakeEnv("CHISEL_AUTH_DIR", credsDir)
+	defer restore()
+
+	confFile := filepath.Join(credsDir, "credentials")
+	contents := ""
+	for _, info := range archive.ProArchiveInfo {
+		contents += fmt.Sprintf("machine %s login foo password bar\n", info.BaseURL)
 	}
+	err := os.WriteFile(confFile, []byte(contents), 0600)
+	c.Assert(err, IsNil)
 
-	_, err = archive.Open(&options)
-	c.Assert(err, ErrorMatches, `.*\bno Ubuntu section`)
+	for pro, info := range archive.ProArchiveInfo {
+		s.base = info.BaseURL
+		s.prepareArchiveAdjustRelease("jammy", "22.04", "amd64", []string{"main", "universe"}, setLabel(info.Label))
+
+		options := archive.Options{
+			Label:      "ubuntu",
+			Version:    "22.04",
+			Arch:       "amd64",
+			Suites:     []string{"jammy"},
+			Components: []string{"main", "universe"},
+			CacheDir:   c.MkDir(),
+			Pro:        pro,
+			PubKeys:    []*packet.PublicKey{s.pubKey},
+		}
+
+		_, err = archive.Open(&options)
+		c.Assert(err, IsNil)
```

Comment:
Let's also check that for non-pro archives, we are not setting the credentials as HTTP header.